inference-engine/src/vpu/graph_transformer/src/frontend/detect_network_batch.cpp

```diff
@@ -83,93 +73,80 @@ void FrontEnd::detectNetworkBatch(
         }
 
         inputShapes[ieData->getName()] = ieShapes;
     }
 
+    if (inputBatch == 1) {
+        env.log->trace("Network is batch 1. Not need to reshape");
+        return;
+    }
+
+    if (inputBatch == 0) {
+        env.log->trace("Unable to decide on network batch size.");
+        return;
+    }
+
     //
     // Special case for DetectionOutput.
     //
 
-    if (!network.getFunction()) {
-        for (auto it = ie::details::CNNNetworkIterator(&network); it != ie::details::CNNNetworkIterator(); ++it) {
-            const auto& layer = *it;
-
-            if (!cmp(layer->type, "DetectionOutput")) {
-                continue;
-            }
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->name);
-
-            if (layer->outData.empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->name);
-            }
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
-            if (!getInputTo(layer->outData.front()).empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration : it is not a network output", layer->name);
-            }
-
-            // 2. Don't support if there multiple outputs as well
-            if (outputsInfo.size() != 1) {
-                VPU_THROW_FORMAT("Unsupported network configuration : layer %s must be the only output of the network", layer->name);
-            }
-
-            model->attrs().set<bool>("withDetectionOutput", true);
-        }
-    } else {
-        const auto layers = network.getFunction()->get_ops();
-        for (const auto& layer : layers) {
-            if (layer->get_type_name() != std::string("DetectionOutput"))
-                continue;
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->get_name());
-
-            if (layer->get_output_size() == 0)
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->get_name());
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
-            for (const auto& outputHandle : layer->outputs()) {
-                for (const auto& inputHandle : outputHandle.get_target_inputs()) {
-                    auto outNode = inputHandle.get_node();
-                    if (dynamic_cast<::ngraph::opset3::Result *>(outNode)) {
-                        continue;
-                    }
-                    VPU_THROW_FORMAT("Unsupported layer %s configuration : it is not a network output", layer->get_name());
-                }
-            }
-
-            // 2. Don't support if there multiple outputs as well
-            if (outputsInfo.size() != 1) {
-                VPU_THROW_FORMAT("Unsupported network configuration : layer %s must be the only output of the network", layer->get_name());
-            }
-            model->attrs().set<bool>("withDetectionOutput", true);
+    const auto layers = network.getFunction()->get_ops();
```

Comment:
Do I understand correctly, that if user sets `forceDeprecatedCnnConversion`, then conversion will happen and here you got nullptr dereference?

Reply:
Do you think it's better to get `ordered_ops` and go from the end, since there is a condition for early exit in case of `DetectionOutput`?

Reply:
if you force conversion then function will exit on the check above. Actually I would expect that if forceDeprecatedCnnConversion is set then model compilation fails anyway

Reply:
This way I'm going to unconditionally fail if this option is set during configurations options refactoring